src/sentry/eventstream/snuba.py

```diff
@@ -161,6 +161,24 @@ def insert(
         event_type = self._get_event_type(event)
         occurrence_data = self._get_occurrence_data(event)
 
+        # instead of normalizing and doing custom 'contexts' processing in snuba, we elect to do it here instead to
+        # avoid having to clutter up snuba code with business logic
+        if event_type == EventStreamEventType.Generic:
+            if event_data.get("contexts"):
+                event_data = dict(event_data)
```

Comment:
do we need to cast this? `event_data` should be a dict from `get_raw_data`

Reply:
Yeah, its mostly a typing since `get_raw_data()` returns a `Mapping` instead of a `MutableMapping`, maybe better to `cast` or do a `deepcopy`